sidecar/internal/service/reclaimspace.go

```
@@ -61,22 +70,50 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get pv: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "failed to get pv %q", pvName)
+	}
+
+	if pv.Spec.CSI == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "pv %q is not a CSI volume", pvName)
+	}
+
+	volID := pv.Spec.CSI.VolumeHandle
+	volAttributes := pv.Spec.CSI.VolumeAttributes
 
 	csiReq := &csiReclaimSpace.ControllerReclaimSpaceRequest{
-		VolumeId:   "",
-		Parameters: map[string]string{},
-		Secrets:    map[string]string{},
+		VolumeId:   volID,
+		Parameters: volAttributes,
 	}
-	res, err := rs.controllerClient.ControllerReclaimSpace(ctx, csiReq)
+
+	// FIXME: use ControllerPublishSecret instead, but it is not set in the PV
+	if pv.Spec.CSI.NodeStageSecretRef != nil {
+		// Get the secrets from the k8s cluster
+		csiReq.Secrets, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
+		if err != nil {
+			klog.Errorf("Failed to get secret: %v", err)
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	csiRes, err := rs.controllerClient.ControllerReclaimSpace(ctx, csiReq)
 	if err != nil {
 		return nil, err
 	}
+	if csiRes == nil {
+		return nil, errors.New("nil value returned as the response of ControllerReclaimSpace")
```

this should return a gRPC error, use `status.Error(...)` instead